### ISSUE TITLE
[no-sequences]: allowInParentheses ain't work with arrows

### ISSUE THREAD

**Issue by bricss** [eslint/eslint#14263]
* **ESLint Version:** 7.23 + @babel/eslint-parser
* **Node Version:** 15.12.0
* **npm Version:** 7.6.3
* **Operating System:** win10_x64

It looks like newly added `allowInParentheses` option of **no-sequences** rule won't work with arrow functions.

```js
const wat = (op) => (console.log('wat'), op);
```

**What did you expect to happen?**
Expect it to pass validation

**What actually happened?**
Validation error: Unexpected use of comma operator

**Comment by mdjermanovic** [comment]
Hi @bricss, thanks for the issue!

> It looks like newly added `allowInParentheses` option of **no-sequences** rule won't work with arrow functions.
> 
> ```js
> const wat = (op) => (console.log('wat'), op);
> ```

This works as expected. `allowInParentheses: true` (default) retains the previous behavior of this rule, which is to allow sequence expressions that are explicitly wrapped in parentheses. For arrow function's body, "explicitly" was considered as being parenthesized twice (ref https://github.com/eslint/eslint/issues/6082):

```js
/* eslint no-sequences: ["error", { "allowInParentheses": true } ] */

const wat1 = (op) => (console.log('wat'), op); // error

const wat2 = (op) => ((console.log('wat'), op)); // ok

```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXNlcXVlbmNlczogW1wiZXJyb3JcIiwgeyBcImFsbG93SW5QYXJlbnRoZXNlc1wiOiB0cnVlIH0gXSAqL1xuXG5jb25zdCB3YXQxID0gKG9wKSA9PiAoY29uc29sZS5sb2coJ3dhdCcpLCBvcCk7IC8vIGVycm9yXG5cbmNvbnN0IHdhdDIgPSAob3ApID0+ICgoY29uc29sZS5sb2coJ3dhdCcpLCBvcCkpOyAvLyBva1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

I've verified that ESLint v7.22.0 with the provided example gives the same result.

Marked as accepted issue to improve documentation. We should add some examples with arrow functions to clarify this.

**Comment by bricss** [comment]
Isn't it like a buttery butter 🧈 to wrap it twice in parentheses? 🤔
